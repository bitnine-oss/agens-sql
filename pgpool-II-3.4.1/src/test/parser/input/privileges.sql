GRANT SELECT, INSERT, UPDATE, DELETE, RULE, REFERENCES, TRIGGER on t1, t2 to x;
GRANT SELECT, INSERT, UPDATE, DELETE, RULE, REFERENCES, TRIGGER on t1, t2 to x WITH GRANT OPTION;
GRANT SELECT, INSERT, UPDATE, DELETE, RULE, REFERENCES, TRIGGER on t1, t2 to GROUP x WITH GRANT OPTION;
GRANT SELECT, INSERT, UPDATE, DELETE, RULE, REFERENCES, TRIGGER on t1, t2 to PUBLIC;
GRANT ALL on t1, t2 to x;
GRANT CREATE on DATABASE db1, db2 TO x,y;
GRANT TEMPORARY on DATABASE db1, db2 TO x,y;
GRANT TEMP on DATABASE db1, db2 TO x,y;
GRANT EXECUTE on FUNCTION func(int, text) TO x,y;
GRANT USAGE ON LANGUAGE lang TO x, y;
GRANT CREATE ON SCHEMA s TO x, y;
GRANT USAGE ON SCHEMA s TO x, y;
GRANT CREATE ON TABLESPACE tblspc TO x, y;
GRANT role1 to role2;
REVOKE SELECT,INSERT,UPDATE,DELETE,RULE,REFERENCES,TRIGGER on t1, t2 FROM x, y;
REVOKE SELECT,INSERT,UPDATE,DELETE,RULE,REFERENCES,TRIGGER on t1, t2 FROM x, y CASCADE;
REVOKE SELECT,INSERT,UPDATE,DELETE,RULE,REFERENCES,TRIGGER on t1, t2 FROM x, y RESTRICT;
REVOKE ALL on t1, t2 FROM x, y;
REVOKE ALL on t1, t2 FROM PUBLIC;
REVOKE GRANT OPTION FOR ALL on t1, t2 FROM PUBLIC;
REVOKE GRANT OPTION FOR CREATE on DATABASE db1,db2 FROM x,y;
REVOKE CREATE on DATABASE db1,db2 FROM x,y;
REVOKE TEMPORARY on DATABASE db1,db2 FROM x,y;
REVOKE TEMP on DATABASE db1,db2 FROM x,y;
REVOKE EXECUTE on FUNCTION foo(int,text) db1,db2 FROM x,y;
REVOKE USAGE ON LANGUAGE lang FROM x,y;
REVOKE CREATE ON SCHEMA s1,s2 FROM x,y;
REVOKE USAGE ON SCHEMA s1,s2 FROM x,y;
REVOKE CREATE ON TABLESPACE tblspc FROM x,y;
REVOKE ADMIN OPTION FOR role1, role2 FROM x,y;
REVOKE role1, role2 FROM x,y;
