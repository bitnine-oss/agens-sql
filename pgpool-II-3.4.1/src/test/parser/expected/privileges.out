GRANT select, insert, update, delete, rule, references, trigger ON "t1","t2" TO "x"
GRANT select, insert, update, delete, rule, references, trigger ON "t1","t2" TO "x" WITH GRANT OPTION
GRANT select, insert, update, delete, rule, references, trigger ON "t1","t2" TO "x" WITH GRANT OPTION
GRANT select, insert, update, delete, rule, references, trigger ON "t1","t2" TO PUBLIC
GRANT ALL ON "t1","t2" TO "x"
GRANT create ON DATABASE "db1", "db2" TO "x","y"
GRANT temporary ON DATABASE "db1", "db2" TO "x","y"
GRANT temp ON DATABASE "db1", "db2" TO "x","y"
GRANT execute ON FUNCTION "func"(int4,"text") TO "x","y"
GRANT usage ON LANGUAGE "lang" TO "x","y"
GRANT create ON SCHEMA "s" TO "x","y"
GRANT usage ON SCHEMA "s" TO "x","y"
GRANT create ON TABLESPACE "tblspc" TO "x","y"
GRANT "role1" TO "role2"
REVOKE select, insert, update, delete, rule, references, trigger ON "t1","t2" FROM "x","y"
REVOKE select, insert, update, delete, rule, references, trigger ON "t1","t2" FROM "x","y" CASCADE
REVOKE select, insert, update, delete, rule, references, trigger ON "t1","t2" FROM "x","y"
REVOKE ALL ON "t1","t2" FROM "x","y"
REVOKE ALL ON "t1","t2" FROM PUBLIC
REVOKE GRANT OPTION FOR ALL ON "t1","t2" FROM PUBLIC
REVOKE GRANT OPTION FOR create ON DATABASE "db1", "db2" FROM "x","y"
REVOKE create ON DATABASE "db1", "db2" FROM "x","y"
REVOKE temporary ON DATABASE "db1", "db2" FROM "x","y"
REVOKE temp ON DATABASE "db1", "db2" FROM "x","y"
syntax error: REVOKE EXECUTE on FUNCTION foo(int,text) db1,db2 FROM x,y;
REVOKE usage ON LANGUAGE "lang" FROM "x","y"
REVOKE create ON SCHEMA "s1", "s2" FROM "x","y"
REVOKE usage ON SCHEMA "s1", "s2" FROM "x","y"
REVOKE create ON TABLESPACE "tblspc" FROM "x","y"
REVOKE ADMIN OPTION FOR "role1", "role2" FROM "x", "y"
REVOKE "role1", "role2" FROM "x", "y"
